src/core/thread/network_data_tlvs.hpp

```diff
@@ -223,7 +223,7 @@ class HasRouteEntry
      */
     void SetPreference(int8_t aPrf)
     {
-        mFlags = (mFlags & ~kPreferenceMask) | ((aPrf << kPreferenceOffset) & kPreferenceMask);
+        mFlags = (mFlags & ~kPreferenceMask) | ((static_cast<uint8_t>(aPrf) << kPreferenceOffset) & kPreferenceMask);
```

Comment:
shall we assert `aPrf` not equal to `10b`, which is defined *Reserved - MUST NOT be sent*, and should be treat as `00b` if received.

Reply:
@bukepo, updated with additional checks. Please review.